### DISCUSSION TITLE
Reading 2D temperature field

### DISCUSSION THREAD

**Discussion by bazharz** [AMReX-Combustion/PeleLMeX#218] [General]
Can the PeleLMeX be configured to incorporate a 2D temperature field from a .dat file for the Flamesheet case?

**Comment by esclapez** [comment]
Do you mean as an initial or a boundary condition ?

**Comment by bazharz** [comment]
As an initial condition!

**Comment by esclapez** [comment]
It is possible to do something like that, but nothing out-of-the-box, you'll will need to implement the IO in the `pelelm_prob*` files. For example in the `Exec/Case/ChallengeProblem` case, we load velocities data from a binary/ASCII file to use them as initial conditions. You'll need the following:
 - define a container for the data you want to load, similar to [these pointers ](https://github.com/AMReX-Combustion/PeleLMeX/blob/be5d1a2b1a3b01d9ce97b6e000236633ae548d5d/Exec/Cases/ChallengeProblem/pelelm_prob_parm.H#L45)
 - Read the data out of the ASCII file, allocate the pointers declared above and copy the data in, as [done here](https://github.com/AMReX-Combustion/PeleLMeX/blob/be5d1a2b1a3b01d9ce97b6e000236633ae548d5d/Exec/Cases/ChallengeProblem/pelelm_prob.cpp#L147)
 - finally, use the data in pelelm_initdata(), [interpolating on the cartesian grid from your dataset](https://github.com/AMReX-Combustion/PeleLMeX/blob/be5d1a2b1a3b01d9ce97b6e000236633ae548d5d/Exec/Cases/ChallengeProblem/pelelm_prob.H#L98)
 
The example of the HIT file is pretty lengthy, but it can handle interpolating the data consistently over multiple AMR during the initialization and so on. You might be able to simplify the code a bit if you know that the resolution of your ASCII data matches the AMR level cartesian grid for instance.

**Comment by bazharz** [comment]
I appreciate your response, and I would like to conclude that this matter can now be considered resolved.
